controller_manager/test/test_load_controller.cpp

```diff
@@ -83,11 +88,113 @@ TEST_F(TestLoadController, load2_known_controller)
+  controller_manager::ControllerSpec abstract_test_controller1 =
+    cm_->get_loaded_controllers()[0];
+
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED,
+    abstract_test_controller1.c->get_current_state().id());
+
+  EXPECT_EQ(
+    cm_->configure_controller(controller_name1), controller_interface::return_type::SUCCESS);
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE,
+    abstract_test_controller1.c->get_current_state().id());
+
+  { //  Start controller
+    RCLCPP_INFO(
+      cm_->get_logger(),
+      "Starting stopped controller");
+    std::vector<std::string> start_controllers = {controller_name1};
+    std::vector<std::string> stop_controllers = {};
+
+    // First activation not possible because controller not configured
+    auto switch_future = std::async(
+      std::launch::async,
+      &controller_manager::ControllerManager::switch_controller, cm_,
+      start_controllers, stop_controllers,
+      STRICT, true, rclcpp::Duration(0, 0));
+
+    ASSERT_EQ(
+      std::future_status::timeout,
+      switch_future.wait_for(std::chrono::milliseconds(100))) <<
+      "switch_controller should be blocking until next update cycle";
+    cm_->update();
+    EXPECT_EQ(
+      controller_interface::return_type::SUCCESS,
+      switch_future.get()
+    );
+
+    ASSERT_EQ(
+      lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE,
+      abstract_test_controller1.c->get_current_state().id());
+  }
+  // Can not configure active controller
+  EXPECT_EQ(cm_->configure_controller(controller_name1), controller_interface::return_type::ERROR);
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE,
+    abstract_test_controller1.c->get_current_state().id());
+
+  {  // Stop controller
+    std::vector<std::string> start_controllers = {};
+    std::vector<std::string> stop_controllers = {controller_name1};
+    RCLCPP_INFO(
+      cm_->get_logger(),
+      "Stopping started controller");
+    auto switch_future = std::async(
+      std::launch::async,
+      &controller_manager::ControllerManager::switch_controller, cm_,
+      start_controllers, stop_controllers,
+      STRICT, true, rclcpp::Duration(0, 0));
+
+    ASSERT_EQ(
+      std::future_status::timeout,
+      switch_future.wait_for(std::chrono::milliseconds(100))) <<
+      "switch_controller should be blocking until next update cycle";
+    cm_->update();
+    EXPECT_EQ(
+      controller_interface::return_type::SUCCESS,
+      switch_future.get()
+    );
+
+    ASSERT_EQ(
+      lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE,
+      abstract_test_controller1.c->get_current_state().id());
+  }
+  // Can not configure inactive controller
+  EXPECT_EQ(cm_->configure_controller(controller_name1), controller_interface::return_type::ERROR);
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE,
+    abstract_test_controller1.c->get_current_state().id());
+}
+
 TEST_F(TestLoadController, update)
```

Comment:
This test does not do anything new than the previous tests, can this be deleted?

Reply:
Agreed, maybe it was an old test and got superseded by newer tests and never removed.